### PR TITLE
[Merged by Bors] - introduce EntityLocation::INVALID const and adjust Entities::get comment

### PR DESCRIPTION
**Pull request by shuoli84**
also remove one un-needed mut.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/entity/mod.rs [hunk 1]
```diff
@@ -563,7 +563,8 @@ impl Entities {
         self.len = 0;
     }
 
-    /// Returns `Ok(Location { archetype: Archetype::invalid(), index: undefined })` for pending entities.
+    /// Returns the location of an [`Entity`].
+    /// Note: for pending entities, returns `Some(EntityLocation::INVALID)`.
```
Comment (JoJoJet) [inline crates/bevy_ecs/src/entity/mod.rs:567]: This could probably be more specific about what a pending entity is.

**Review by JoJoJet — APPROVED** [review]

File: crates/bevy_ecs/src/entity/mod.rs (hunk 1 above)
Comment (shuoli84) [reply crates/bevy_ecs/src/entity/mod.rs:567]: I think struct's doc comment is a better place to document this.

File: crates/bevy_ecs/src/entity/mod.rs [hunk 2]
```diff
@@ -771,6 +767,15 @@ pub struct EntityLocation {
     pub table_row: TableRow,
 }
 
+impl EntityLocation {
+    const INVALID: EntityLocation = EntityLocation {
```
Comment (james7132) [inline crates/bevy_ecs/src/entity/mod.rs:771]: This (and EntityMeta::EMPTY) deserves a doc comment on how it's used, but I won't block on it.

**Review by james7132 — APPROVED** [review]

File: crates/bevy_ecs/src/entity/mod.rs (hunk 2 above)
Comment (shuoli84) [reply crates/bevy_ecs/src/entity/mod.rs:778]: I added a doc for empty, pending, reserved and invalid entity, which I don't sure whether is accurate. Mind take a look?

File: crates/bevy_ecs/src/entity/mod.rs [hunk 3]
```diff
@@ -2,6 +2,11 @@
 //!
 //! An **entity** exclusively owns zero or more [component] instances, all of different types, and can dynamically acquire or lose them over its lifetime.
 //!
+//! **empty entity**: Entity ownes zero [component] is called empty entity, it has a valid [`EntityLocation`] which points to [`ArchetypeId::EMPTY`]
+//! **pending entity**: Entity reserved, but not flushed yet([`Entities::flush`] docs for reference).
+//! **reserved entity**: same as **pending entitiy**.
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/entity/mod.rs:7]:
```suggestion
//! **reserved entity**: same as **pending entity**.
```

File: crates/bevy_ecs/src/entity/mod.rs [hunk 4]
```diff
@@ -2,6 +2,11 @@
 //!
 //! An **entity** exclusively owns zero or more [component] instances, all of different types, and can dynamically acquire or lose them over its lifetime.
 //!
+//! **empty entity**: Entity ownes zero [component] is called empty entity, it has a valid [`EntityLocation`] which points to [`ArchetypeId::EMPTY`]
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/entity/mod.rs:5]:
```suggestion
//! **empty entity**: Entity with zero components. Has a valid [`EntityLocation`] which points to [`ArchetypeId::EMPTY`].
```

File: crates/bevy_ecs/src/entity/mod.rs [hunk 5]
```diff
@@ -2,6 +2,11 @@
...
+//! **pending entity**: Entity reserved, but not flushed yet([`Entities::flush`] docs for reference).
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/entity/mod.rs:6]:
```suggestion
//! **pending entity**: Entity reserved, but not flushed yet(see [`Entities::flush`] docs for reference).
```

Comment (JoJoJet) [reply crates/bevy_ecs/src/entity/mod.rs:6]: Also please add a space before the parentheses.

**Comment by shuoli84** [comment]
@james7132 ping

File: crates/bevy_ecs/src/entity/mod.rs [hunk 6]
```diff
@@ -2,6 +2,11 @@
...
+//! **reserved entity**: same as **pending entitiy**.
+//! **invalid entity**: **pending entity** flushed with location [`EntityLocation::INVALID`]([`Entities::flush_as_invalid`] docs for reference).
+//!
```
[the remark this answers is not in the document]
Comment (shuoli84) [reply crates/bevy_ecs/src/entity/mod.rs:9]: Removed EntityLocation related info, but keeps a brief desc for these terms.

**Review by james7132 — APPROVED** [review]
bors r+

**Pull request closed without merging**